### ISSUE TITLE
Crash @llvm::isAArch64FrameOffsetLegal

### ISSUE THREAD

**Issue by glandium** [llvm/llvm-project#55187]
Testcase: [ecp_secp384r1-6e751c.c.gz](https://github.com/llvm/llvm-project/files/8589166/ecp_secp384r1-6e751c.c.gz)
Command: `"clang" "-cc1" "-triple" "aarch64-unknown-linux-gnu" "-emit-obj" "-O2" "-std=gnu99" "-x" "c" "/tmp/ecp_secp384r1-6e751c.c"`
```
Stack dump:
0.	Program arguments: ./bin/clang -cc1 -triple aarch64-unknown-linux-gnu -emit-obj -O2 -std=gnu99 -x c /tmp/ecp_secp384r1-6e751c.c
1.	<eof> parser at end of file
2.	Code generation
3.	Running pass 'Function Pass Manager' on module '/tmp/ecp_secp384r1-6e751c.c'.
4.	Running pass 'Prologue/Epilogue Insertion & Frame Finalization' on function '@fiat_secp384r1_mul'
 #0 0x00007f369089d830 PrintStackTraceSignalHandler(void*) Signals.cpp:0:0
 #1 0x00007f369089b18c SignalHandler(int) Signals.cpp:0:0
 #2 0x00007f368f93bd60 (/lib/x86_64-linux-gnu/libc.so.6+0x3bd60)
 #3 0x00007f3692c630bb llvm::isAArch64FrameOffsetLegal(llvm::MachineInstr const&, llvm::StackOffset&, bool*, unsigned int*, long*) (/tmp/llvm/obj/bin/../lib/libLLVM-15git.so+0x2fb10bb)
 #4 0x00007f3692c632df llvm::rewriteAArch64FrameIndex(llvm::MachineInstr&, unsigned int, unsigned int, llvm::StackOffset&, llvm::AArch64InstrInfo const*) (/tmp/llvm/obj/bin/../lib/libLLVM-15git.so+0x2fb12df)
 #5 0x00007f3692c8f62d llvm::AArch64RegisterInfo::eliminateFrameIndex(llvm::MachineInstrBundleIterator<llvm::MachineInstr, false>, int, unsigned int, llvm::RegScavenger*) const (/tmp/llvm/obj/bin/../lib/libLLVM-15git.so+0x2fdd62d)
 #6 0x00007f3690e1cde8 (anonymous namespace)::PEI::replaceFrameIndices(llvm::MachineBasicBlock*, llvm::MachineFunction&, int&) PrologEpilogInserter.cpp:0:0
 #7 0x00007f3690e1f5d5 (anonymous namespace)::PEI::replaceFrameIndices(llvm::MachineFunction&) PrologEpilogInserter.cpp:0:0
 #8 0x00007f3690e2595d (anonymous namespace)::PEI::runOnMachineFunction(llvm::MachineFunction&) PrologEpilogInserter.cpp:0:0
 #9 0x00007f3690ce87e4 llvm::MachineFunctionPass::runOnFunction(llvm::Function&) (/tmp/llvm/obj/bin/../lib/libLLVM-15git.so+0x10367e4)
#10 0x00007f3690a05750 llvm::FPPassManager::runOnFunction(llvm::Function&) (/tmp/llvm/obj/bin/../lib/libLLVM-15git.so+0xd53750)
#11 0x00007f3690a058bc llvm::FPPassManager::runOnModule(llvm::Module&) (/tmp/llvm/obj/bin/../lib/libLLVM-15git.so+0xd538bc)
#12 0x00007f3690a063f1 llvm::legacy::PassManagerImpl::run(llvm::Module&) (/tmp/llvm/obj/bin/../lib/libLLVM-15git.so+0xd543f1)
#13 0x00007f36986d6c50 clang::EmitBackendOutput(clang::DiagnosticsEngine&, clang::HeaderSearchOptions const&, clang::CodeGenOptions const&, clang::TargetOptions const&, clang::LangOptions const&, llvm::StringRef, llvm::Module*, clang::BackendAction, std::unique_ptr<llvm::raw_pwrite_stream, std::default_delete<llvm::raw_pwrite_stream> >) (/tmp/llvm/obj/bin/../lib/libclang-cpp.so.15git+0x1a22c50)
#14 0x00007f3698ab43dc clang::BackendConsumer::HandleTranslationUnit(clang::ASTContext&) (/tmp/llvm/obj/bin/../lib/libclang-cpp.so.15git+0x1e003dc)
#15 0x00007f3697656e29 clang::ParseAST(clang::Sema&, bool, bool) (/tmp/llvm/obj/bin/../lib/libclang-cpp.so.15git+0x9a2e29)
#16 0x00007f3699305209 clang::FrontendAction::Execute() (/tmp/llvm/obj/bin/../lib/libclang-cpp.so.15git+0x2651209)
#17 0x00007f3699287feb clang::CompilerInstance::ExecuteAction(clang::FrontendAction&) (/tmp/llvm/obj/bin/../lib/libclang-cpp.so.15git+0x25d3feb)
#18 0x00007f3699380920 clang::ExecuteCompilerInvocation(clang::CompilerInstance*) (/tmp/llvm/obj/bin/../lib/libclang-cpp.so.15git+0x26cc920)
#19 0x000055ba40f6eaec cc1_main(llvm::ArrayRef<char const*>, char const*, void*) (./bin/clang+0x14aec)
#20 0x000055ba40f6b9c9 ExecuteCC1Tool(llvm::SmallVectorImpl<char const*>&) driver.cpp:0:0
#21 0x000055ba40f67f8e main (./bin/clang+0xdf8e)
#22 0x00007f368f926d0a __libc_start_main ./csu/../csu/libc-start.c:308:16
#23 0x000055ba40f6b37a _start (./bin/clang+0x1137a)
```
This is a regression from 81904454f7cdebecedb1185d8112b630a7124350.

**Comment by glandium** [comment]
Cc: @Kmeakin

**Comment by glandium** [comment]
Fixed by d29fc6e7d24f9bf67512180c0be3f08526049a27
